engine/src/proxy.rs

```diff
@@ -335,12 +337,12 @@ mod test {
             let put = WriteOperation::new_put("kv", "key2".into(), "value2".into());
             assert!(engine.write_batch(vec![put], false).is_ok());
 
-            let mut buf = vec![0u8; snapshot.size().numeric_cast()];
+            let mut buf = BytesMut::with_capacity(snapshot.size().numeric_cast());
             snapshot.read_exact(&mut buf).await.unwrap();
 
             buf.extend([0u8; 100]); // add some padding, will be ignored when receiving
 
-            received_snapshot.write_all(&buf).await.unwrap();
+            received_snapshot.write_all(buf.freeze()).await.unwrap();
```

Comment:
Please refer to https://github.com/tokio-rs/bytes/issues/401 for a better understanding of `freeze` function. It's not zero-cost. We should reconsider whether to use `bytes` or not.

Reply:
There's a commit that fix this problem https://github.com/tokio-rs/bytes/commit/05e9d5cab95e5ce1d57dd498a9567639de50f841. Also since we already filled the buf, calls to `Vec::shrink_to_fit()` will not cause additional allocation.